leadbutt.py

```diff
@@ -144,18 +162,19 @@ def leadbutt(config_file, cli_options, verbose=False, **kwargs):
             # we need a copy of the metric dict with the MetricName swapped out
             this_metric = metric.copy()
             this_metric['MetricName'] = metric_name
-            results = conn.get_metric_statistics(
-                period_local,  # minimum: 60
-                start_time,
-                end_time,
-                metric_name,  # RequestCount, CPUUtilization
-                metric['Namespace'],  # AWS/ELB, AWS/EC2
-                metric['Statistics'],  # Sum, Maximum
+            results = get_metric_statistics(
+                connection=conn,
+                period=period_local,
+                start_time=start_time,
+                end_time=end_time,
+                metric_name=metric_name,
+                namespace=metric['Namespace'],
+                statistics=metric['Statistics'],
                 dimensions=metric['Dimensions'],
-                unit=unit,
+                unit=unit
             )
-            # sys.stderr.write('{} {}\n'.format(options['Count'], len(results)))
             output_results(results, this_metric, options)
+            time.sleep(float(kwargs.get('interval', 0)) / 1000.0)
```

Comment:
The overall logic is complicating enough that one has to read the code 2-3 times. The dual usage of `-i`, while very clever, will catch someone off guard. Let's comment this.